code/game/machinery/doors/airlock.dm

```diff
@@ -197,6 +196,36 @@
 	if(id_tag)
 		id_tag = "[port.shuttle_id]_[id_tag]"
 
+/obj/machinery/door/airlock/proc/rebuild_parts()
+	if(part_overlays)
+		vis_contents -= part_overlays
+		QDEL_LIST(part_overlays)
+	else
+		part_overlays = list()
+	var/list/parts_desc = params2list(anim_parts)
+	for(var/part_id in parts_desc)
+		var/obj/effect/overlay/airlock_part/P = new
+		P.side_id = part_id
+		var/list/open_offset = splittext(parts_desc[part_id], ",")
+		P.open_px = text2num(open_offset[1])
+		P.open_py = text2num(open_offset[2])
+		if(open_offset.len >= 3)
+			P.move_start_time = text2num(open_offset[3])
+		if(open_offset.len >= 4)
+			P.move_end_time = text2num(open_offset[4])
+		if(open_offset.len >= 5)
+			P.aperture_angle = text2num(open_offset[5])
+		vis_contents += P
+		part_overlays += P
+		P.icon = icon
+		P.icon_state = part_id
+		P.parent = src
+		P.name = name
+	if(mask_filter)
+		filters -= mask_filter
+	mask_filter = filter(type="alpha",icon=mask_file,x=mask_x,y=mask_y)
+	filters += mask_filter
```

Comment:
do not use raw filter manipulation, we have an api wrapped around this you should be using (alpha_mask_filter, add_filter, remove_filter)